eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/Argument.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.core.Box;
+import com.schedjoules.eventdiscovery.framework.serialization.core.Key;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.Present;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public class Argument<T> extends AbstractCachingOptional<T>
+{
+    // TODO Worth implementing here, not using the abstract class
```

Comment:
It comes back to the same question, I think, whether we agree on a contract that all Fragments has to have an argument.
Just because of convenient usage:
```
// Convenient:
Event event = new Argument(Keys.EVENT, getArguments()).value();
// Non-convenient:
Event event = new Argument(Keys.EVENT, new NullSafe<>(getArguments())).value();
```
I think we should avoid having the need to write down the second one, so either we keep null check in `Argument` (either explicit or using `NullSafe`) or we introduce a contract to always have argument for Fragment.
Right?